beluga_ros/include/beluga_ros/particle_cloud.hpp

```diff
@@ -81,10 +81,36 @@ struct almost_equal_to<Sophus::SE2<Scalar>> {
 
   /// Compares `a` and `b` for near equality.
   bool operator()(const Sophus::SE2<Scalar>& a, const Sophus::SE2<Scalar>& b) const {
-    using std::abs, std::atan, std::tan;
+    using std::abs;
     return (abs(a.translation().x() - b.translation().x()) < linear_resolution) &&
            (abs(a.translation().y() - b.translation().y()) < linear_resolution) &&
-           (abs(atan(tan(a.so2().log() - b.so2().log()))) < angular_resolution);
+           (abs((a * b.inverse()).log().z()) < angular_resolution);
+  }
+
+  const Scalar linear_resolution;   ///< Resolution for translational coordinates, in meters.
+  const Scalar angular_resolution;  ///< Resolution for rotational coordinates, in radians.
+};
+
+/// std::equal_to equivalent specialized for SE(3) types, with user resolutions.
+template <typename Scalar>
+struct almost_equal_to<Sophus::SE3<Scalar>> {
+  /// Constructs near equality functor.
+  /**
+   * \param _linear_resolution Resolution for translational coordinates, in meters.
+   * \param _angular_resolution Resolution for rotational coordinates, in radians.
+   */
+  explicit almost_equal_to(Scalar _linear_resolution, Scalar _angular_resolution)
+      : linear_resolution(_linear_resolution), angular_resolution(_angular_resolution) {}
+
+  /// Compares `a` and `b` for near equality.
+  bool operator()(const Sophus::SE3<Scalar>& a, const Sophus::SE3<Scalar>& b) const {
+    using std::abs;
+    const Eigen::Vector<Scalar, 6> angles = (a * b.inverse()).log();
+    return (abs(a.translation().x() - b.translation().x()) < linear_resolution) &&
+           (abs(a.translation().y() - b.translation().y()) < linear_resolution) &&
+           (abs(a.translation().z() - b.translation().z()) < linear_resolution) &&
+           (abs(angles[3]) < angular_resolution) && (abs(angles[4]) < angular_resolution) &&
+           (abs(angles[5]) < angular_resolution);
```

Comment:
@pvela2017 that is,
```suggestion
    using std::abs;
    const Sophus::SE3<Scalar> diff = a * b.inverse();
    return (abs(diff.translation().x()) < linear_resolution) &&
           (abs(diff.translation().y()) < linear_resolution) &&
           (abs(diff.translation().z()) < linear_resolution) &&
           (abs(diff.so3().angleX()) < angular_resolution) &&
           (abs(diff.so3().angleY()) < angular_resolution) &&
           (abs(diff.so3().angleZ()) < angular_resolution);
```
It's similar in 2D.

Reply:
Update in 442f27f5235d31816a0cdeb67b85ad0671a9c406